cabot/cabotapp/models.py

```diff
@@ -473,14 +481,55 @@ def last_result(self):
     def should_run(self):
         '''Returns true if the check should run, false otherwise.'''
 
-        # Do not run if the activity counter is enabled and zero.
-        # - If the DB entry does not exist, we assume a value of zero.
+        # Handle special cases for activity-counted checks, which may have run delays
         if self.use_activity_counter:
             counters = ActivityCounter.objects.filter(status_check=self)
-            if len(counters) == 0 or counters[0].count <= 0:
-                logger.info("Skipping check '{}', activity counter is zero".format(self.name))
+
+            # If the check's activity counter doesn't exist, the check should not run
+            if len(counters) == 0:
+                return False
+
+            counter = counters[0]
```

Comment:
This is nit-picky, but consider:
```python
try:
  counter = ActivityCounter.objects.get(status_check=self)
except ActivityCounter.DoesNotExist:
  return False
```
Gets rid of pulling the first item from the array.